### DISCUSSION TITLE
Не срабатывает Multiarc

### DISCUSSION THREAD

**Discussion by kmserg1** [elfmz/far2l#1983] [Q&A]
Не могу понять где проблема. При попытке создать архив через Shift-F1 появляется окно терминала, на котором видно команду, которая должна выполниться: 
~$ "/usr/local/bin/far2l" --libexec "/usr/local/lib/far2l/Plugins/multiarc/plug/multiarc.far-plug-mb" "/home/kmserg" BuiltinMain libarch A:zip React-Total.zip -@   -- "React-Total/"
Больше ничего не происходит. far2l висит, пока я не нажму Alt-Control-C. Тогда команда прерывается, а архив так и не создается.
При этом из-под root все срабатывает нормально. Папку ~/.config/far2l чистил. Безрезультатно. Куда еще смотреть?

**Comment by elfmz** [comment]
Для начала методом перебора попробовать определить конкретеую поддиректорию в /home/kmserg которая вызывает проблему и далее углубляясь внутрь - может даже и конкретный файл

**Comment by kmserg1** [comment]
Нашел причину и устранил. Хотя и не понял почему так присходит.
Причина была в том, что у меня в файле .bashrc была строчка duf,
котороая при открытии консоли выполняла duf. 
duf показывает заполненность дисков в удобном виде.

Убрал duf - все заработало.

Но почему это происходит - непонятно. 
Ведь duf - это просто красивый вариант df.

**Comment by elfmz** [comment]
поставил себе duf и вписал его в конец ~/.bashrc - не воспроизвелось..

**Comment by kmserg1** [comment]
Загадка.
Только что перепроверил. Подтверждаю: у меня наличие duf в .bashrc вызывает зависание far2l при попытке архивации через Shift-F1 и вообще при попытке выполнить любую команду в командной строке. Только что пропрбовал находясь в far2l набрать и выполнить "cat 1.txt" и получил такое же зависание.
Добавил duf в .bashrc для root и у него началось такое же поведение.

Для информации: 
far2l: 2.5.3-2024-01-28-bcd0ae95-beta

duf: 0.8.1-1+b6

root@homecomp:~# cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"

root@homecomp:~# uname -a
Linux homecomp 6.1.0-17-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.69-1 (2023-12-30) x86_64 GNU/Linux

**Comment by elfmz** [comment]
а этот duf не может при каких то условиях интерактивно чтото спрашивать у пользователя?

**Comment by kmserg1** [comment]
Я заметил, что для того, чтобы .bashrc сработал для far2l не нужно перезагружать far2l. Достаточно внести измение в .bashrc и оно уже действует. 
То есть .bashrc перечитыватся перед каждым выполнением консольной команды в far2l. И по идее, каждый раз должна выполняться duf.
Я не нашел, чтобы duf как-то общался с пользователем. Он просто отрабатывает и заканчивается.
Но может far2l  не любит, чтобы в его терминал что-то выводили?
Или duf что-то меняет в окружении? 
Проверил ERRORLEVEl - все нормально, он 0.

**Comment by kmserg1** [comment]
И в терминале far2l не видно результатов выполнения duf. Возможно, что по какой-то причине зависает duf и не может опросить диски?

**Comment by akruphi** [comment]
> Только что пропрбовал находясь в far2l набрать и выполнить "cat 1.txt" и получил такое же зависание. 

Вот это ^^^ наиболее странное. А простой `ls` у Вас внутри far2l отрабатывает?

Данные ошибки у Вас проявляются как в far2l GUI, так и `far2l --tty`?

**Comment by kmserg1** [comment]
ls точно так же зависает. 
far2l --tty точно так же зависает. 
![screen-2024-02-03-22-46-18](https://github.com/elfmz/far2l/assets/12230824/546bf645-a387-4e9a-bdf5-e3d8d2bb55c4)
Все таки мне кажется, что при выполнении консольной команды не может выполниться duf из .bashrc

**Comment by akruphi** [comment]
Очень странно. Заинтриговали что же именно у Вас ломается добавлением в .bashrc.
А проверьте, если не сложно, при закомментированном duf добавление в .bashrc только `df` или только `ls --color=auto` приводят к глюку - т.е. проблема из-за обращений к файловым системам или из-за раскрашивания вывода в терминале или из-за символов рамок, выводимых duf.

**Comment by kmserg1** [comment]
Добавление в .bashrc 
а) только df - не приводит к зависаниям 
б) только ls --color=auto - не приводит к зависаниям
в) проверил также du - не приводит к зависаниям 

Надо наверно смотреть текст программы duf. Что она и как делает.
Но тут уж я не специалист. Могу написать автору duf.
